create-custom-map-style/README.md

```diff
@@ -0,0 +1,471 @@
+      "scripts": {
+        "test": "echo \"Error: no test specified\" && exit 1"
+      },
+      "keywords": [],
+      "author": "",
+      "license": "ISC"
+    }
+    ```
+
+1. Next we need to add the Amazon Location Service API as a data source. However, in order to do this we need to use a local proxy that can handle SigV4 (TL;DR a way to provide your AWS credentials with http requests). For this we will use [tessera](https://github.com/mojodna/tessera) and [tilelive-aws](https://github.com/beatleboy501/tilelive-aws). From a new terminal (the `maputnik` process will be occupying the other terminal) run the following:
+
+    ```bash
+
+    nvm use 15
+    npm i tessera tilelive-aws
+    ```
+
+1. Presuming you have the standard AWS environment variables (`AWS_ACCESS_KEY_ID`, `AWS_SECRET_ACCESS_KEY`, `AWS_REGION`) set correctly, this should be enough to get started using `tessera`. You'll need to provide the full file path to your module, so from your terminal run:
+
+    ```bash
+
+    node_modules/tessera/bin/tessera -r tilelive-aws aws:///<YOUR AMAZON LOCATION SERVICE MAP NAME>
+    ```
+
+    You can also add this as a script to your `package.json` file to save time on subsequent invocations.
+
+1. You should see some output in sdtout like `Listening at http://0.0.0.0:8080`.
+
+1. Now go back to Maputnik in your browser and click "Data Sources" in the top navigation bar. There should not be any active data sources. If you see any be sure to delete them.
+
+    ![Data Sources](media/MaputnikDataSources.png)
+
+    *Data Sources*
+
+1. We're going to add a new data source as shown below.
+
+- `Source ID`: `esri`
+- `Source Type`: `Vector (XYZ URLs)`
+- `1st Tile URL`: `http://localhost:8080/{z}/{x}/{y}.pbf`
+- `Min Zoom`: `0`
+- `Max Zoom`: `22`
+
+    ![New Data Source](media/MaputnikNewSource.png)
+
+    *New Data Source*
+
+1. In addition to Tiles, the type of web map we will be styling also makes use of **Glyphs** and **Sprites**. We can't use Tessera to proxy the endpoints for these, but we can however use some out of the box endpoints built in to Maputnik. Sprites: `https://maputnik.github.io/osm-liberty/sprites/osm-liberty` and Glyphs: `https://api.maptiler.com/fonts/{fontstack}/{range}.pbf?key={key}`)
+
+1. If you want to download a copy of the Sprites and Glyphs your Amazon Location Service Map uses we've included a helper script `downloadSpritesGlyphs.sh` :
+
+    ```bash
+    ./downloadSpritesGlyphs.sh <YOUR AMAZON LOCATION SERVICE MAP NAME> <YOUR AWS CLI PROFILE NAME>
+    ```
+  
+This should create a folder for `sprites` and a folder for `glyphs` with a number of options for each. [Sprites](https://docs.aws.amazon.com/cli/latest/reference/location/get-map-sprites.html) can be either `.json` or `.png` files with an optional `@2x` version (Retina-quality images). [Glyphs](https://docs.aws.amazon.com/cli/latest/reference/location/get-map-glyphs.html) are a combination of a font-family and a Unicode range.
```

Comment:
It's not that sprites can be either JSON or PNG. Spritesheets consist of a paired set where the PNG contains all of the images and the JSON file provides the offsets of each image within the PNG.